chain/chunks/src/lib.rs

```diff
@@ -449,13 +449,29 @@ impl ShardsManager {
             || self.runtime_adapter.will_care_about_shard(account_id, parent_hash, shard_id, is_me)
     }
 
+    /// Send requests for a chunk
+    /// Depending on whether the node tracks the shard, it either downloads the full chunk
+    /// or only parts that it concerns, specifically, the part that should be owned by the node if
+    /// it is a validator, plus the parts that are sent to the shards that the node is tracking
+    ///
+    /// `height`: block height of the chunk
+    /// `parent_hash`: hash of the prev block of the block that includes the requested chunk
+    /// `shard_id`: shard id of the requested chunk
+    /// `chunk_hash`: chunk hash of the requested chunk
+    /// `force_request_full`: if it is true, the node will request the full chunk regardless whether
+    ///    it is tracking the shard
+    /// `request_from_single_source`: if it is true, the node requests all parts from a single node
+    ///                               instead of sending one request to each part owner
+    /// `request_own_parts_from_others`: whether we should request parts owned by us
+    /// `request_from_archival`: whether to request from archival nodes
     fn request_partial_encoded_chunk(
         &mut self,
         height: BlockHeight,
         parent_hash: &CryptoHash,
         shard_id: ShardId,
         chunk_hash: &ChunkHash,
         force_request_full: bool,
+        request_from_single_source: bool,
         request_own_parts_from_others: bool,
         request_from_archival: bool,
```

Comment:
I think it makes sense to create an enum here to reduce the number of arguments and potentially make the code cleaner